node/Makefile

```diff
@@ -145,7 +145,11 @@ LDFLAGS=-ldflags "\
 	-X $(PACKAGE_NAME)/buildinfo.BuildDate=$(DATE) \
 	-X $(PACKAGE_NAME)/buildinfo.GitRevision=$(GIT_COMMIT)"
 
-SRC_FILES=$(shell find ./pkg -name '*.go')
+# Source golang files on which compiling the calico-node binary depends.
+SRC_FILES=$(shell find ./pkg -name '*.go') \
+	  $(shell find ../felix -name '*.go') \
```

Comment:
I just merged a change in felix that makes it dependent on *.h in felix/bpf. Its is possible that we only change one of the cgo files and that should also trigger a rebuild. It is rare, but we just hit it as an issue.

Why don't we do something like `make -C ../felix build` ?

Reply:
I don't really want to have to call into another Makefile because that can't be properly represented as a dependency as far as I know - i.e., you'll need to run that target every time.

Maybe it works OK if felix has proper dependency tracking so that multiple calls to build won't rebuild the code? We can also add .h files here though - we can include as many types of files as we need.

That said, I believe changes to ".h" files should be caught by the dependency tracking for the REMOTE_DEPS targets already, so probably don't need to do anything special

Reply:
This Makefile should only depend on REMOTE_DEPS and those should be updated by building respective components like felix or typha

Reply:
SRC_FILES is simply listing to golang files within this repo that the calico-node binary (not image) build depends on. The calico-node binary build **does not require felix, libcalico-go or confd, etc to be rebuilt** so it shouldn't be calling into those directories to build anything. However we should rebuild the calico-node binary if golang files in those repos are modified because we import the source code (not compiled artifacts) here. 

Maybe I should rename REMOTE_DEPS - they're really not remote any more since they're in this repo.

Reply:
Maybe it would be simpler and more accurate to just put everything into SRC_FILES. Yeah, I like that.